db/dcrpg/pgblockchain.go

```diff
@@ -66,6 +67,59 @@ func (d *DevFundBalance) Balance() *explorer.AddressBalance {
 	return d.AddressBalance
 }
 
+// ticketPoolDataCache stores the most recent ticketpool graphs information
+// fetched to minimize the possibility of making multiple queries to the db
+// fetching the same information.
+type ticketPoolDataCache struct {
+	sync.RWMutex
+	Height uint64
+	// BarGraphsCache persists data for the Ticket purchase distribution chart
+	// and Ticket Price Distribution chart
+	BarGraphsCache map[dbtypes.ChartGrouping][]*dbtypes.PoolTicketsData
+	// DonutGraphCache persist data for the Number of tickets outputs pie chart.
+	DonutGraphCache map[dbtypes.ChartGrouping]*dbtypes.PoolTicketsData
+}
+
+// ticketPoolGraphsCache persists the latest ticketpool data queried from the db.
+var ticketPoolGraphsCache = &ticketPoolDataCache{
+	BarGraphsCache:  make(map[dbtypes.ChartGrouping][]*dbtypes.PoolTicketsData),
+	DonutGraphCache: make(map[dbtypes.ChartGrouping]*dbtypes.PoolTicketsData),
+}
+
+// TicketPoolData is a thread-safe way to access the ticketpool graphs data
+// stored in the cache.
+func TicketPoolData(interval dbtypes.ChartGrouping, height uint64) (barGraphs []*dbtypes.PoolTicketsData,
+	donutChart *dbtypes.PoolTicketsData, actualHeight uint64, isFound bool) {
+	ticketPoolGraphsCache.RLock()
+	defer ticketPoolGraphsCache.RUnlock()
+
+	if ticketPoolGraphsCache.Height != height {
+		return nil, nil, 0, false
+	}
+
+	barGraphs, isFound = ticketPoolGraphsCache.BarGraphsCache[interval]
+	donutChart = ticketPoolGraphsCache.DonutGraphCache[interval]
```

Comment:
since you are using `isFound` to check if the data is stale then you could have:
`isFound = ticketPoolGraphsCache.Height == height`
rather than what is there.

Reply:
or maybe change the variable name to `isStale`..